velox/expression/tests/ArgumentTypeFuzzer.cpp

```diff
@@ -48,7 +48,24 @@ void ArgumentTypeFuzzer::determineUnboundedTypeVariables() {
     // Random randomType() never generates unknown here.
     // TODO: we should extend randomType types and exclude unknown based
     // on variableInfo.
-    bindings_[variableName] = randType();
+    // Respect orderable/comparable constraint.
+    TypePtr typePtr = nullptr;
+    while (true) {
```

Comment:
This may take a very long time... We need to put a limit on the number of iterations or find a better way altogether. 

Perhaps, create a custom version of randType that skips MAPs.

```
TypePtr randType(
    FuzzerGenerator& rng,
    const std::vector<TypePtr>& scalarTypes,
    int maxDepth) {
  const int numScalarTypes = scalarTypes.size();
  // Should we generate a scalar type?
  if (maxDepth <= 1 || rand<bool>(rng)) {
    return scalarTypes[rand<uint32_t>(rng) % numScalarTypes];
  }
  switch (rand<uint32_t>(rng) % 2) {
    case 0:
      return ARRAY(randType(rng, scalarTypes, maxDepth - 1));
    default:
      return randRowType(rng, scalarTypes, maxDepth - 1);
  }
}
```

Reply:
Got it, thanks. I will revert this commit and file another PR adding a test according to your guidance.

Reply:
@duanmeng I'll be out till next Wed. 

Our custom function may look like this.

```
TypePtr randOrderableType(
    FuzzerGenerator& rng,
    int maxDepth = 5) {
  // Should we generate a scalar type?
  if (maxDepth <= 1 || rand<bool>(rng)) {
    return randType(rng, 0);
  }

  // ARRAY or ROW?
  if (rand<bool>(rng)) {
      return ARRAY(randOrderableType(rng, maxDepth - 1));
  } 

  int numFields = 1 + rand<uint32_t>(rng) % 7;
  std::vector<std::string> names;
  std::vector<TypePtr> fields;
  for (int i = 0; i < numFields; ++i) {
    names.push_back(fmt::format("f{}", i));
    fields.push_back(randOrderableType(rng, maxDepth - 1));
  }
  return ROW(std::move(names), std::move(fields));  
}
```